templates/cicd/pipeline_cfn.yml

```
@@ -188,7 +188,11 @@ Resources:
               - codestar-connections:PassConnection
               - codestar-connections:PassedToService
             Resource:
+              {{- if eq .Source.ConnectionARN ""}}
               - !Ref SourceConnection
+              {{- else}}
+              - {{$.Source.Connection}}
+              {{- end}}
```

If they provide us the connection name maybe we can instead reconstruct the ARN here like this:
```suggestion
              {{- else}}
              - !Sub 'arn:aws:codestar-connections:${AWS::Region}:${AWS::AccountId}:connection/{{$.Source.Connection}}'
              {{- end}}
```

---

Hm, but these connection ARNs aren't made up of the connection names. 
They're like this: arn:aws:codestar-connections:us-east-1:241234567890:connection/601f96ce-4eb3-4010-8dce-45d2f4990b6c

---

ooooh okay! I didn't know that, it makes sense why they would do that! 
Could we possibly use the app session to get the ARN of the connection while running `pipeline update`? maybe through (https://docs.aws.amazon.com/codestar-connections/latest/APIReference/API_ListConnections.html)

---

I like that idea! OR...what are pros/cons of passing it as a flag rather than making it a manifest field?

---

If it's in the manifest it's easy to repeat it, like if they delete the pipeline and want to re-create it it's v easy to do through the manifest